### PR TITLE
Remove NaN values in spatial_rate_map

### PR DESCRIPTION
**Pull request by dragly**
Otherwise, numpy.digitize will return indices that are outside the bins.

### CODE REVIEW

**Comment by dragly** [comment]
This is perhaps something we want to add in a general function that can be used in other places.

@simetenn You've had some experience with removing NaNs in UncertainPy. Does this look good to you?

File: exana/tracking/fields.py [hunk 1]
```diff
@@ -35,6 +35,17 @@ def spatial_rate_map(x, y, t, sptr, binsize=0.01*pq.m, box_xlen=1*pq.m,
     out : rate map, xbins, ybins
     """
     from exana.misc.tools import is_quantities
+
+    if np.isnan(x).any() or np.isnan(y).any() or np.isnan(z).any():
```
Comment (simetenn) [inline exana/tracking/fields.py]: I assume that np.isnan(z) -> np.isnan(t).

File: exana/tracking/fields.py [hunk 2]
```diff
@@ -35,6 +35,17 @@ def spatial_rate_map(x, y, t, sptr, binsize=0.01*pq.m, box_xlen=1*pq.m,
...
+        print("WARNING: spatial_rate_map called with NaN values in x, y or t arrays. "
+              "These will be removed.")
+
+        # the & is the logical and operator, and ~ is the logical not operator
+        nan_indices = np.isnan(x) & np.isnan(y) & np.isnan(t)
```
Comment (simetenn) [inline exana/tracking/fields.py:44]: This should work as long as x, y, and t are numpy arrays of type float/int and not None, numpy arrays of type object, or float/ints (i think?).

**Review by simetenn — CHANGES_REQUESTED** [review]

File: exana/tracking/fields.py (hunk 1 above)
Comment (dragly) [reply exana/tracking/fields.py]: that's correct :)

File: exana/tracking/fields.py (hunk 2 above)
Comment (dragly) [reply exana/tracking/fields.py:44]: x, y and t should be quantities arrays of type float

**Pull request closed without merging**